pkg/sdn/plugin/common.go

```diff
@@ -31,20 +31,60 @@ func clusterNetworkToString(n *osapi.ClusterNetwork) string {
 	return fmt.Sprintf("%s (network: %q, hostSubnetBits: %d, serviceNetwork: %q, pluginName: %q)", n.Name, n.Network, n.HostSubnetLength, n.ServiceNetwork, n.PluginName)
 }
 
+//determine if an ip address (ipaddr) is contained within a list of cidrs
+func cidrListContains(cidrList []*net.IPNet, ipaddr net.IP) (*net.IPNet, bool) {
+	for _, cidr := range cidrList {
+		if cidr.Contains(ipaddr) {
+			return cidr, true
+		}
+	}
+	return nil, false
+}
+
+func clusterNetworkListContains(clusterNetworks []ClusterNetwork, ipaddr net.IP) (*net.IPNet, bool) {
+	for _, cn := range clusterNetworks {
+		if cn.ClusterCIDR.Contains(ipaddr) {
+			return cn.ClusterCIDR, true
+		}
+	}
+	return nil, false
+}
+
 type NetworkInfo struct {
-	ClusterNetwork *net.IPNet
-	ServiceNetwork *net.IPNet
+	ClusterNetworks []ClusterNetwork
+	ServiceNetwork  *net.IPNet
 }
 
-func parseNetworkInfo(clusterNetwork string, serviceNetwork string) (*NetworkInfo, error) {
-	cn, err := netutils.ParseCIDRMask(clusterNetwork)
-	if err != nil {
-		_, cn, err := net.ParseCIDR(clusterNetwork)
+type ClusterNetwork struct {
+	ClusterCIDR      *net.IPNet
+	HostSubnetLength uint32
+}
+
+//determine if two cidr addresses intersect
+func intersect(cidr1, cidr2 *net.IPNet) bool {
+	return cidr2.Contains(cidr1.IP) || cidr1.Contains(cidr2.IP)
+}
+
+func parseNetworkInfo(clusterNetwork []osapi.ClusterNetworkEntry, serviceNetwork string) (*NetworkInfo, error) {
+	var cns []ClusterNetwork
+
+	for _, entry := range clusterNetwork {
+		clusterAddress, err := netutils.ParseCIDRMask(entry.CIDR)
 		if err != nil {
-			return nil, fmt.Errorf("failed to parse ClusterNetwork CIDR %s: %v", clusterNetwork, err)
+			_, clusterAddress, err := net.ParseCIDR(entry.CIDR)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse ClusterNetwork CIDR %s: %v", clusterAddress, err)
+			}
+			glog.Errorf("Configured clusterNetworkCIDR value %q is invalid; treating it as %q", entry.CIDR, clusterAddress.String())
 		}
-		glog.Errorf("Configured clusterNetworkCIDR value %q is invalid; treating it as %q", clusterNetwork, cn.String())
+		for _, cn := range cns {
+			if intersect(cn.ClusterCIDR, clusterAddress) {
+				return nil, fmt.Errorf("Two of the cidr addresses overlap: %s, %s", cn.ClusterCIDR.String(), clusterAddress.String())
+			}
+		}
+		cns = append(cns, ClusterNetwork{ClusterCIDR: clusterAddress, HostSubnetLength: entry.HostSubnetLength})
 	}
+
```

Comment:
You are already checking for overlaps among clusternetowrk cidrs here, then this should be used in ValidateClusterNetwork()?